### DISCUSSION TITLE
Add object type field

### DISCUSSION THREAD

**Discussion by ryanleichty** [craftcms/cms#15626] [Ideas] [answered]
This is an idea coming from Sanity. In Sanity, they have an [object type schema](https://www.sanity.io/docs/object-type) that allows you to nest other schema types/fields within it. You can optionally collapse the object field too. This helps keep things organized as well as frees up handle names.

In a way, it's kind of similar to matrix fields, except you're not adding nested entries. Just nested fields.

![image](https://github.com/user-attachments/assets/6750f6e5-38e1-4691-a132-a2e95f57a4bc)

Possible UI idea:

![CleanShot 2024-08-29 at 09 52 12@2x](https://github.com/user-attachments/assets/83028dfa-1db3-4b8d-a5cf-7ba257990dd6)

We're currently handling this by naming things like `seoTitle`, `seoDescription`, and `seoImage`:

![CleanShot 2024-08-29 at 09 53 38@2x](https://github.com/user-attachments/assets/12ff46ad-5bbf-4f06-bda0-e0d1dd12122e)

**Comment by brandonkelly** [comment]
Thanks for the suggestion! This is basically the same idea as #14600 – something we’re starting to think through 😉

**Reply by ryanleichty** [reply]
Nice, yeah that's basically the same idea 👍

**Comment by alexanderbuergin** [comment]
What if we could define reusable Field Group Types, similar to Entry Types?

You’d create one group — e.g. «SEO Fields» with Title, Description, Image, Robots — and reuse it across entry types or matrix blocks. Same for common setups like «Hero Section» or «Date Range».

This would avoid repeating the same set of fields over and over, improve maintainability, and keep things consistent across the project.

Think of it as modular field modeling: define once, reuse anywhere.

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.8 is out, which introduces a new “Content Block” field type, which can be used to create reusable nested field layouts. (#17424)
